src/main/java/com/three_tech_solutions/slot_app/components/payment_processors/factory/PaymentProcessorFactory.java

```diff
@@ -0,0 +1,25 @@
+package com.three_tech_solutions.slot_app.components.payment_processors.factory;
+
+import com.three_tech_solutions.slot_app.components.payment_processors.PaymentProcessor;
+import com.three_tech_solutions.slot_app.components.payment_processors.implementations.DiaEspecifoPaymentProccesor;
+import com.three_tech_solutions.slot_app.components.payment_processors.implementations.PrincipioDeMesPaymentProcessor;
+import com.three_tech_solutions.slot_app.data.enums.PlanType;
+import org.springframework.stereotype.Component;
+import java.util.Map;
+
+@Component
+public class PaymentProcessorFactory {
+
+    Map<PlanType, PaymentProcessor> paymentProcessors;
```

Comment:
Acá nos falto el método de acceso y declarar que es constante


```suggestion
    private final Map<PlanType, PaymentProcessor> paymentProcessors;
```